client/question/services/code_evaluators/PythonPrereqCheckService.js

```diff
@@ -53,6 +55,33 @@ tie.factory('PythonPrereqCheckService', [
       });
     };
 
+    /**
+     * Processes studentCode and converts all single quote + double quote str
+     * to filler character "x".
+     * This is to preserve character length for error checks.
+     *
+     * @param {string} [code] The code to convert
+     * @return {string} code where all 'foo' & "bar" converted to xxxxx & xxxxx
+     */
+    var obscureStringCharacters = function(code) {
+      var regexp = new RegExp(/'([^']*)'|"([^"]*)"/, 'g');
+      var matches = [];
+      var obscureCode = '';
+      var pointer = 0;
+
+      while ((matches = regexp.exec(code)) !== null) {
```

Comment:
This is a bit clumsy IMO; consider looking into https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace